org.sofproject.alsa.topo/src/org/sofproject/alsa/topo/model/AlsaTopoGraph.java

```diff
@@ -536,7 +536,56 @@ public void serialize() throws CoreException, IOException {
 		}
 		outputFile.getParent().refreshLocal(1, null);
 	}
+	
+	@Override
+	public String getPipelineString() {
+		try {
+			ByteArrayOutputStream os = new ByteArrayOutputStream();
+			Writer writer = new BufferedWriter(new OutputStreamWriter(os));
+			// tlv-s
+			tlvs.serialize(writer);
+
+			// vendor tokens
+			vTokensIndex.serialize(writer);
+
+			// vendor tuples
+			vTuplesIndex.serialize(writer);
```

Comment:
These comments are here just to decrypt variable name. Better just name them `vendorTokens/Tuples...` etc